supabase/functions/stats/index.ts

```diff
@@ -13,9 +14,54 @@ const failActions = [
   'update_fail',
   'download_fail',
 ]
+
+// ios sends 13 fields while android sends 11 fields
+const jsonRequestSchema = z.object({
+  app_id: z.string({
+    required_error: 'App ID is required',
+    invalid_type_error: 'App ID name must be a string',
+  }),
+  device_id: z.string({
+    required_error: 'Device ID is required',
+    invalid_type_error: 'Device ID must be a string',
+  }).max(36),
+  platform: z.string({
+    required_error: 'Platform type is required',
+    invalid_type_error: 'Platform type must be a string',
+  }),
+  version_name: z.string({
+    required_error: 'Version name is required',
+    invalid_type_error: 'Version name must be a string',
+  }),
+  version_os: z.string({
+    required_error: 'Version OS is required',
+    invalid_type_error: 'Version OS must be a string',
+  }),
+  version_code: z.optional(z.string()),
+  version_build: z.optional(z.string()),
+  action: z.optional(z.string()),
+  custom_id: z.optional(z.string()),
+  channel: z.optional(z.string()),
+  plugin_version: z.optional(z.string()),
+  is_emulator: z.boolean().default(false),
+  is_prod: z.boolean().default(true),
+}).refine(data => reverseDomainRegex.test(data.app_id), {
+  message: 'App ID name must be a reverse domain string',
+}).refine(data => deviceIdRegex.test(data.device_id), {
+  message: 'Device ID must be a valid UUID string',
+})
+
 async function main(url: URL, headers: BaseHeaders, method: string, body: AppStats) {
   try {
     console.log('body', body)
+    let _parseResult = null
+    try {
+      _parseResult = jsonRequestSchema.parse(body)
+    }
+    catch (error) {
+      return sendRes({ error: `Cannot parse json: ${error}` }, 400)
```

Comment:
This error response will look like unreadable garbage, zod errors are not great to parse to json as far as I remember from my testing. Could you provide an example of a failed parse request?

Reply:
A failed parse typically looks like this in the console:
```bash
serving the request with /home/deno/functions/updates
[Info] Listening on http://localhost:9999/

[Error] sendRes error {
  "error": "Cannot parse json: [\n  {\n    \"code\": \"invalid_type\",\n    \"expected\": \"string\",\n    \"received\": \"undefined\",\n    \"path\": [\n      \"app_id\"\n    ],\n    \"message\": \"App ID is required\"\n  }\n]"
}

[Error] sendRes error "{\"error\":\"Cannot parse json: [\\n  {\\n    \\\"code\\\": \\\"invalid_type\\\",\\n    \\\"expected\\\": \\\"string\\\",\\n    \\\"received\\\": \\\"undefined\\\",\\n    \\\"path\\\": [\\n      \\\"app_id\\\"\\n    ],\\n    \\\"message\\\": \\\"App ID is required\\\"\\n  }\\n]\"}"
```

Since the intent behind the use of Zod validation is to mitigate abuse from bad actors as per https://github.com/Cap-go/capgo/issues/384 this response will serve its purpose (by slowing them down). 

An unmodified Capgo plugin should still be able to consume the `/update` and `/stats` endpoints as before so there should be no impact to existing users.